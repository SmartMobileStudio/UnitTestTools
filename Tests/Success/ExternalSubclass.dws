uses
  ECMA.Console;

asm
  function TExternal() {
     this.Field = "hello";
  }
  TExternal.prototype.GetIt = function () { return this.Field };
  TExternal.prototype.PrintIt = function () { Console.Log(this.GetIt()) };
end;
               
type
  TExternal = class external
    Field: String;
    function GetIt: String;
    procedure PrintIt;
  end;

type
  TSub = class (TExternal)
    function GetIt: String;
  end;

type 
  TSub2 = class (TExternal) 
  end;

function TSub.GetIt: String;
begin
  Result := Field + ' world';
end;   

var base := new TExternal;
base.printIt;

var sub := new TSub;
sub.printit;

if (sub is TExternal) then Console.Log('ok1');
if (sub is TSub) then Console.Log('ok2');

if (sub is TSub2) then Console.Log('bug1');

sub := nil;

if (sub is TSub) then Console.Log('bug2');