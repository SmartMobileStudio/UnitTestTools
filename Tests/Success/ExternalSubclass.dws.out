SmartMS command-line compiler, (c) 2016 The Smart Company AS
Building Tests\Success\ExternalSubclass.dws
[INFO] Building project ''...
[INFO] Compiling...
Warning: Incompatible types: "TSub" and "TSub2" [line: 42, column: 9]
[INFO] Generating JavaScript...
[INFO] Linking...
